preprocessing/nextclade/src/loculus_preprocessing/prepro.py

```diff
@@ -725,16 +703,25 @@ def run(config: Config) -> None:
         if config.nextclade_dataset_name:
             download_nextclade_dataset(dataset_dir, config)
         total_processed = 0
+        etag = ""
         while True:
             logging.debug("Fetching unprocessed sequences")
-            unprocessed = parse_ndjson(fetch_unprocessed_sequences(config.batch_size, config))
-            if len(unprocessed) == 0:
+            etag, unprocessed = fetch_unprocessed_sequences(etag, config)
+            if not unprocessed:
                 # sleep 1 sec and try again
                 logging.debug("No unprocessed sequences found. Sleeping for 1 second.")
                 time.sleep(1)
                 continue
+            # Reset etag if we have new data, only use if we have no new data
```

Comment:
I am not quite sure why you reset this here?
The new etag will be set in 
```
etag, unprocessed = fetch_unprocessed_sequences(etag, config)
```
when there is a change in data. If anything maybe we should reset the etag to "" when there has been no change in e.g. 10min?

Reply:
This is a good question, I'm pretty sure I wrapped my head around this and it made sense at the time but I clearly need to improve the comments then if even I can't immediately answer your question.